pkg/pillar/cmd/nim/nim.go

```diff
@@ -170,7 +170,8 @@ func Run(ps *pubsub.PubSub) {
 	nimCtx.PubDevicePortConfigList = pubDevicePortConfigList
 	nimCtx.PubDeviceNetworkStatus = pubDeviceNetworkStatus
 	dnc := &nimCtx.DeviceNetworkContext
-	devicenetwork.IngestPortConfigList(dnc)
+	removeLegacy := true // XXX should only be set when we are updating EVE
```

Comment:
The comment indicates otherwise? Aren't we setting this always? Do you plan to fix it later?

Reply:
Yeah I'm waiting for the eve version update code cominhg in with the GlobalConfigPR hopefully soon, so that we can use the same check here to only do this when doing an eveimage-update.

Reply:
No longer needed since once we have currentIndex=0 will drop all of the other entires